lib/crawler/output_sink/elasticsearch.rb

```diff
@@ -128,6 +137,14 @@ def pipeline_params
         @pipeline_params ||= DEFAULT_PIPELINE_PARAMS.merge(es_config[:pipeline_params] || {}).deep_stringify_keys
       end
 
+      def lock_queue
+        @queue_locked = true
+      end
+
+      def unlock_queue
+        @queue_locked = false
+      end
```

Comment:
Should definitely be atomic operations + raise if something went wrong (tried locking while already locked, tried unlocking when already unlocked)